src/NUnitFramework/framework/Attributes/RandomAttribute.cs

```diff
@@ -258,14 +283,36 @@ public override IEnumerable GetData(IParameterInfo parameter)
 
                 _randomizer = Randomizer.GetRandomizer(parameter.ParameterInfo);
 
+                if (Distinct && _inRange && !CanBeDistinct(_min, _max, _count))
+                    throw new NUnitException($"Random values can't be distinct. Method: \"{parameter.Method}\"; Parameter: \"{parameter.ParameterInfo.Name}\"");
```

Comment:
I'm always in favor of failures as soon and loud as possible when the end user asks for something inherently contradictory.